File: app/src/main/java/com/forkmang/fragment/Table_OrdersListing_Fragment.java
package com.forkmang.fragment;

import static com.forkmang.helper.Constant.TOKEN_LOGIN;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ProgressBar;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.forkmang.R;
import com.forkmang.adapter.TableOrdersListing_Adapter;
import com.forkmang.data.TableOrderListing;
import com.forkmang.helper.Constant;
import com.forkmang.helper.StorePrefrence;
import com.forkmang.helper.Utils;
import com.forkmang.network_call.Api;
import com.google.gson.Gson;
import com.google.gson.JsonObject;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class Table_OrdersListing_Fragment extends Fragment {
    RecyclerView recyclerView;
    StorePrefrence storePrefrence;
    ProgressBar progressBar;
    ArrayList<TableOrderListing> tableOrderListingArrayList;
    public static Table_OrdersListing_Fragment instance;


    public static Table_OrdersListing_Fragment newInstance() {
        return new Table_OrdersListing_Fragment();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_booktable_layout_orders, container, false);
        instance= this;
        storePrefrence=new StorePrefrence(getContext());
        progressBar = view.findViewById(R.id.progressBar);
        recyclerView = view.findViewById(R.id.booktable_recycleview_orders);
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        return view;

    }

    @Override
    public void onResume() {
        super.onResume();
        if (Utils.isNetworkAvailable(getContext())) {
            callapi_gettableorderslist();
        }
        else{
            Toast.makeText(getContext(), Constant.NETWORKEROORMSG, Toast.LENGTH_SHORT).show();

        }

    }



    //Api code for Book Table start
    private void callapi_gettableorderslist()
    {
        progressBar.setVisibility(View.VISIBLE);
        Api.getInfo().getbooktable_listing("Bearer "+storePrefrence.getString(TOKEN_LOGIN)).
                enqueue(new Callback<JsonObject>() {
                    @Override
                    public void onResponse(Call<JsonObject> call, Response<JsonObject> response) {
                        try{
                            if(response.code() == Constant.SUCCESS_CODE_n)
                            {
                                JSONObject jsonObject = new JSONObject(new Gson().toJson(response.body()));
                                //Log.d("Result", jsonObject.toString());

                                if(jsonObject.getString("status").equalsIgnoreCase(Constant.SUCCESS_CODE))
                                {
                                    if(jsonObject.getJSONArray("data").length() > 0)
                                    {
                                        TableOrderListing tableOrderListing = new TableOrderListing();
                                        tableOrderListingArrayList = new ArrayList<TableOrderListing>();
                                        for(int i = 0; i< jsonObject.getJSONArray("data").length(); i++)
                                        {
                                            JSONObject mjsonobj = jsonObject.getJSONArray("data").getJSONObject(i);

                                            tableOrderListing.setOrder_id(mjsonobj.getString("id"));
                                            tableOrderListing.setUser_id(mjsonobj.getString("user_id"));
                                            tableOrderListing.setOrder_total(mjsonobj.getString("total"));
                                            tableOrderListing.setPayment_status(mjsonobj.getString("payment_status"));
                                            tableOrderListing.setOrder_status(mjsonobj.getString("order_status"));
                                            tableOrderListing.setResturant_id(mjsonobj.getString("restaurant_id"));

                                            tableOrderListing.setStr_restroname(mjsonobj.getJSONObject("restaurant").getString("rest_name"));
                                            tableOrderListing.setResturant_branch(mjsonobj.getJSONObject("restaurant").getString("rest_branch"));
                                            tableOrderListing.setResturant_contact(mjsonobj.getJSONObject("restaurant").getString("contact"));

                                            if(mjsonobj.isNull(mjsonobj.getString("timing")))
                                            {
                                                tableOrderListing.setResturant_timing("00");
                                            }
                                            else{
                                                tableOrderListing.setResturant_timing(mjsonobj.getString("timing"));
                                            }
                                            tableOrderListingArrayList.add(tableOrderListing);
                                        }

                                        TableOrdersListing_Adapter bookTableAdapter_orders = new TableOrdersListing_Adapter(getActivity(),tableOrderListingArrayList);
                                        recyclerView.setAdapter(bookTableAdapter_orders);
                                        progressBar.setVisibility(View.GONE);

                                    }
                                    else{
                                        //no data in array list
                                        progressBar.setVisibility(View.GONE);
                                        Toast.makeText(getContext(), Constant.NODATA, Toast.LENGTH_LONG).show();
                                    }
                                }
                                else{
                                    progressBar.setVisibility(View.GONE);
                                    Toast.makeText(getContext(), Constant.ERRORMSG, Toast.LENGTH_LONG).show();
                                }
                            }
                            else{
                                progressBar.setVisibility(View.GONE);
                                Toast.makeText(getContext(), Constant.ERRORMSG, Toast.LENGTH_LONG).show();
                            }
                        }
                        catch (JSONException ex)
                        {
                            ex.printStackTrace();
                            progressBar.setVisibility(View.GONE);
                            Toast.makeText(getContext(), Constant.ERRORMSG, Toast.LENGTH_LONG).show();
                        }
                    }
                    @Override
                    public void onFailure(Call<JsonObject> call, Throwable t) {
                        Toast.makeText(getContext(), Constant.ERRORMSG, Toast.LENGTH_LONG).show();
                        progressBar.setVisibility(View.GONE);
                    }
                });
    }



    //Fragment Instance
    public static Table_OrdersListing_Fragment GetInstance()
    {
        return instance;
    }
}

File: app/src/main/java/com/forkmang/fragment/Table_OrdersListing_Fragment.kt
package com.forkmang.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import com.forkmang.adapter.TableOrdersListing_Adapter
import com.forkmang.data.TableOrderListing
import com.forkmang.databinding.FragmentBooktableLayoutOrdersBinding
import com.forkmang.helper.Constant
import com.forkmang.helper.Constant.TOKEN_LOGIN
import com.forkmang.helper.StorePrefrence
import com.forkmang.helper.Utils.isNetworkAvailable
import com.forkmang.helper.showToastMessage
import com.forkmang.network_call.Api.info
import com.google.gson.Gson
import com.google.gson.JsonObject
import org.json.JSONException
import org.json.JSONObject
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class Table_OrdersListing_Fragment : Fragment() {
    var storePrefrence: StorePrefrence? = null
    var tableOrderListingArrayList: ArrayList<TableOrderListing>? = null

    private var _binding: FragmentBooktableLayoutOrdersBinding? = null
    private val binding get() = _binding!!


    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentBooktableLayoutOrdersBinding.inflate(inflater, container, false)
        storePrefrence = StorePrefrence(requireContext())
        binding.booktableRecycleviewOrders.layoutManager = LinearLayoutManager(context)
        return binding.root
    }

    override fun onResume() {
        super.onResume()
        if (isNetworkAvailable(requireContext())) {
            callapi_gettableorderslist()
        } else {
            context?.showToastMessage(Constant.NETWORKEROORMSG)
        }
    }

    //Api code for Book Table start
    private fun callapi_gettableorderslist() {
        binding.progressBar.visibility = View.VISIBLE
        info.getbooktable_listing("Bearer " + storePrefrence?.getString(TOKEN_LOGIN))
            ?.enqueue(object : Callback<JsonObject?> {
                override fun onResponse(call: Call<JsonObject?>, response: Response<JsonObject?>) {
                    try {
                        if (response.code() == Constant.SUCCESS_CODE_n) {
                            val jsonObject = JSONObject(Gson().toJson(response.body()))
                            //Log.d("Result", jsonObject.toString());
                            if (jsonObject.getString("status")
                                    .equals(Constant.SUCCESS_CODE, ignoreCase = true)
                            ) {
                                if (jsonObject.getJSONArray("data").length() > 0) {
                                    val tableOrderListing = TableOrderListing()
                                    tableOrderListingArrayList = ArrayList()
                                    for (i in 0 until jsonObject.getJSONArray("data").length()) {
                                        val mjsonobj =
                                            jsonObject.getJSONArray("data").getJSONObject(i)
                                        tableOrderListing.order_id = mjsonobj.getString("id")
                                        tableOrderListing.user_id = mjsonobj.getString("user_id")
                                        tableOrderListing.order_total = mjsonobj.getString("total")
                                        tableOrderListing.payment_status =
                                            mjsonobj.getString("payment_status")
                                        tableOrderListing.order_status =
                                            mjsonobj.getString("order_status")
                                        tableOrderListing.resturant_id =
                                            mjsonobj.getString("restaurant_id")
                                        tableOrderListing.str_restroname =
                                            mjsonobj.getJSONObject("restaurant")
                                                .getString("rest_name")
                                        tableOrderListing.resturant_branch =
                                            mjsonobj.getJSONObject("restaurant")
                                                .getString("rest_branch")
                                        tableOrderListing.resturant_contact =
                                            mjsonobj.getJSONObject("restaurant")
                                                .getString("contact")
                                        if (mjsonobj.isNull(mjsonobj.getString("timing"))) {
                                            tableOrderListing.resturant_timing = "00"
                                        } else {
                                            tableOrderListing.resturant_timing =
                                                mjsonobj.getString("timing")
                                        }
                                        tableOrderListingArrayList?.add(tableOrderListing)
                                    }
                                    val bookTableAdapter_orders = TableOrdersListing_Adapter(
                                        requireActivity(), tableOrderListingArrayList
                                    )
                                    binding.booktableRecycleviewOrders.adapter =
                                        bookTableAdapter_orders
                                    binding.progressBar.visibility = View.GONE
                                } else {
                                    //no data in array list
                                    binding.progressBar.visibility = View.GONE
                                    context?.showToastMessage(Constant.NODATA)
                                }
                            } else {
                                binding.progressBar.visibility = View.GONE
                                context?.showToastMessage(Constant.ERRORMSG)
                            }
                        } else {
                            binding.progressBar.visibility = View.GONE
                            context?.showToastMessage(Constant.ERRORMSG)
                        }
                    } catch (ex: JSONException) {
                        ex.printStackTrace()
                        binding.progressBar.visibility = View.GONE
                        context?.showToastMessage(Constant.ERRORMSG)
                    }
                }

                override fun onFailure(call: Call<JsonObject?>, t: Throwable) {
                    context?.showToastMessage(Constant.ERRORMSG)
                    binding.progressBar.visibility = View.GONE
                }
            })
    }

    companion object {
        fun newInstance(): Table_OrdersListing_Fragment {
            return Table_OrdersListing_Fragment()
        }
    }
}